control/settings/controllers/settings.home.controller.js

```diff
@@ -57,16 +63,26 @@
           if (newObj.data && newObj.data.settings && newObj.data.settings.mapView == 'hide' && newObj.data.settings.defaultView == 'map') {
             SettingsHome.couponInfo.data.settings.defaultView = 'list';
             SettingsHome.showMessage = true;
-            setTimeout(function(){
+            setTimeout(function () {
               SettingsHome.showMessage = false;
               $scope.$digest();
-            },2000);
+            }, 2000);
           }
-          if(newObj.data) {
+          if (newObj.data && newObj.data.settings && newObj.data.settings.toggleEmployeeCode == 'on') {
+            SettingsHome.showCode = true;
+          }
+
+          if (newObj.data) {
             DataStore.save(newObj.data, tag).then(saveSuccess, SaveError);
           }
         }
 
+        function saveEmployeeCode(newObj) {
+          if (newObj.data && newObj.data.settings && newObj.data.settings.employeeCode != '') {
+            console.log('klik', newObj.data.settings.employeeCode)
```

Comment:
Why do we just have a console.log here? Shouldn't this function fire a request or sthg?

Reply:
It's resolved now, rewrote the logic behind this completely